app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -245,6 +248,31 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
         if (savedInstanceState == null) {
             viewModel.onViewReady()
         }
+
+        lifecycle.addObserver(object : LifecycleObserver {
+            @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+            fun onStop() {
```

Comment:
There is an issue if we only generate the preview image `onStop`. If you add several tabs using the overflow menu and go to the tab switcher, all the tabs created will have a blank preview image because `onStop` never gets called and, when it does, `viewModel.url` is null. But even if `viewModel.url` wasn't null it would only generate the preview for the latest tab. 

Without giving it much thought (haven't thought about the implications of doing it) I think you can potentially move this logic to `onHiddenChanged` to generate the preview every time the fragment is hidden.

Reply:
Nice catch! I like that suggestion; will investigate this

Reply:
I've added the `onHiddenChanged` hook. We still need the `onStop` hook too, so both now co-exist.